kubernetes/istio-auth-15.yaml

```diff
@@ -0,0 +1,342 @@
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+      - name: apiserver
+        image: docker.io/istio/manager:2017-04-24-23.54.11
+        imagePullPolicy: Always
+        args: ["apiserver", "-v", "2"]
+        ports:
+        - containerPort: 8081
+        env:
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+---
+
+################################
+# Istio ingress controller
+################################
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-ingress-controller
+  labels:
+    infra: istio-ingress-controller
+spec:
+  type: LoadBalancer
+  ports:
+  - port: 80
+#   nodePort: 32000
+    name: http
+  selector:
+    infra: istio-ingress-controller
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: istio-ingress-service-account
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: istio-ingress-controller
+spec:
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        alpha.istio.io/sidecar: ignore
+      labels:
+        infra: istio-ingress-controller
+    spec:
+      serviceAccountName: istio-ingress-service-account
+      containers:
+      - name: istio-ingress-controller
+        image: docker.io/istio/proxy:2017-04-24-23.54.11
+        args: ["proxy", "ingress", "-v", "2"]
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 80
+        env:
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+---
+
+################################
+# Istio egress envoy
+################################
+apiVersion: v1
+kind: Service
+metadata:
+  name: istio-egress
+spec:
+  ports:
+  - port: 80
+  selector:
+    app: istio-egress
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: istio-egress
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: istio-egress
+    spec:
+      containers:
+      - name: proxy
+        image: docker.io/istio/proxy:2017-04-24-23.54.11
+        imagePullPolicy: Always
+        args: ["proxy", "egress", "-v", "2"]
+        env:
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+---
+
+# istio-system namespace for Istio CA.
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: istio-system
+---
+# Istio CA in istio-system namespace.
+apiVersion: v1
+kind: Deployment
+apiVersion: extensions/v1beta1
+metadata:
+  name: istio-ca
+  namespace: istio-system
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        istio: istio-ca
+    spec:
+      containers:
+      - name: istio-ca
+        image: docker.io/istio/istio-ca:2017-04-13-17.43.08
+        imagePullPolicy: IfNotPresent
```

Comment:
About using the file in the demo: This file is not used for the demo, but the tutorial. istio-namespace-ca.yaml is used for the demo, which is exactly as you described.

About deleting the secrets in other namespaces: Yes, I think we need to have the caveats saying the certs/keys created for other namespaces won't be deleted. But as long as these secrets are not used, no one will care about it, right? Will there be potential troubles?